src/main/java/core/basesyntax/strategy/impl/BalanceOperationImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class BalanceOperationImpl implements OperationHandler {
+
+    @Override
+    public void applyOperation(FruitTransaction transaction) {
+        if (transaction.getQuantity() < MIN_QUANTITY) {
+            throw new RuntimeException("Balance can`t be negative.Actual: "
+                    + transaction.getQuantity());
+        }
+        Storage.reports.put(transaction.getFruitName(), transaction.getQuantity());
```

Comment:
Direct manipulation of the storage map is not recommended. Consider encapsulating this logic within the `Storage` class by providing a method to update the storage, ensuring that all updates go through a single point of control.